diff --git a/liveusb/gui.py b/liveusb/gui.py
index 7a91dcf..3d95c4f 100755
--- a/liveusb/gui.py
+++ b/liveusb/gui.py
@@ -34,7 +34,7 @@ from datetime import datetime
 from PyQt4 import QtCore, QtGui
 
 from liveusb import LiveUSBCreator, LiveUSBError, LiveUSBInterface, _
-from liveusb.releases import releases, get_fedora_releases
+from liveusb.releases import releases, get_fedora_releases, RFREMIX_RELEASES, FEDORA_RELEASES
 if sys.platform == 'win32':
     from liveusb.urlgrabber.grabber import URLGrabber, URLGrabError
     from liveusb.urlgrabber.progress import BaseMeter
@@ -338,8 +338,7 @@ class LiveUSBDialog(QtGui.QDialog, LiveUSBInterface):
 
     def refresh_releases(self):
         self.live.log.info(_('Refreshing releases...'))
-        global releases
-        releases = get_fedora_releases()
+        releases = get_fedora_releases(RFREMIX_RELEASES) + get_fedora_releases(FEDORA_RELEASES)
         self.downloadCombo.clear()
         for release in [release['name'] for release in releases]:
             self.downloadCombo.addItem(release)
diff --git a/liveusb/releases.py b/liveusb/releases.py
index db37385..cceb828 100644
--- a/liveusb/releases.py
+++ b/liveusb/releases.py
@@ -3,34 +3,215 @@ from urlgrabber import urlread
 from urlgrabber.grabber import URLGrabError
 
 FEDORA_RELEASES = 'http://dl.fedoraproject.org/pub/fedora/linux/releases/'
+RFREMIX_RELEASES = 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/'
 ARCHES = ('armhfp', 'x86_64', 'i686', 'i386')
 
 # A backup list of releases, just in case we can't fetch them.
 fedora_releases = [
-    {'name': 'Fedora 20 x86_64 Desktop',
-     'sha256': 'cc0333be93c7ff2fb3148cb29360d2453f78913cc8aa6c6289ae6823372a77d2',
-     'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/x86_64/Fedora-Live-Desktop-x86_64-20-1.iso'},
-    {'name': 'Fedora 20 x86_64 KDE',
-     'sha256': '08360a253b4a40dff948e568dba1d2ae9d931797f57aa08576b8b9f1ef7e4745',
-     'url': 'https://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/x86_64/Fedora-Live-KDE-x86_64-20-1.iso'},
-    {'name': 'Fedora 20 i686 Desktop',
-     'sha256': 'b115c5653b855de2353e41ff0c72158350f14a020c041462f35ba2a47bd1e33b',
-     'url': 'https://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/i386/Fedora-Live-Desktop-i686-20-1.iso'},
-    {'name': 'Fedora 20 i686 KDE',
-     'sha256': 'd859132ea9496994ccbb5d6e60c9f40ae89ba31f8a4a1a2a883d6d45901de598',
-     'url': 'https://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/i386/Fedora-Live-KDE-i686-20-1.iso'},
-    {'name': 'Fedora 19 x86_64 Desktop',
-     'sha256': '21f0197284b9088b32d683f83a71bd42261f1df885a63b1eb87254d1ca096f12',
-     'url': 'https://dl.fedoraproject.org/pub/fedora/linux/releases/19/Live/x86_64/Fedora-Live-Desktop-x86_64-19-1.iso'},
-    {'name': 'Fedora 19 x86_64 KDE',
-     'sha256': 'f4479d5639f62d7398722c6a432ba94711b80d5011d5b64e7afebb4f4ac10cf7',
-     'url': 'https://dl.fedoraproject.org/pub/fedora/linux/releases/19/Live/x86_64/Fedora-Live-KDE-x86_64-19-1.iso'},
-    {'name': 'Fedora 19 i686 Desktop',
-     'sha256': 'ce9797802ef1f7aa670fffd04f209631c171b8ded5dc26f61df898cb9441c839',
-     'url': 'https://dl.fedoraproject.org/pub/fedora/linux/releases/19/Live/i386/Fedora-Live-Desktop-i686-19-1.iso'},
-    {'name': 'Fedora 19 i686 KDE',
-     'sha256': 'e81717564f96ee0f8c6d8b0186e7fca44da57f2ef154ea427b6d765ea139a083',
-     'url': 'https://dl.fedoraproject.org/pub/fedora/linux/releases/19/Live/i386/Fedora-Live-KDE-i686-19-1.iso'},
+ {'name': 'RFRemix 20 U1 x86_64 GNOME',
+  'sha256': '346873bce0d43e79695c7a92e274c09eb913f39b11158c741e4fab65c534b8c1',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-U1-x86_64-Live-GNOME.iso'},
+ {'name': 'RFRemix 20 U1 x86_64 KDE',
+  'sha256': '41e268bdbc1d8f261bee9c76e97448be704d6a40cc72f05465453f7d45216f48',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-U1-x86_64-Live-KDE.iso'},
+ {'name': 'RFRemix 20 U1 x86_64 KDE minimal',
+  'sha256': 'da71e0919188d7e66ff11b1e6eb331dc52d22d719c99b70934064b933d1db032',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-U1-x86_64-Live-KDE-minimal.iso'},
+ {'name': 'RFRemix 20 U1 x86_64 LXDE',
+  'sha256': 'e5341d725e6e2d65c3998617c3a9cd61455b3430bb5f747d7da71cbb02c75e3d',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-U1-x86_64-Live-LXDE.iso'},
+ {'name': 'RFRemix 20 U1 x86_64 MATE',
+  'sha256': 'f02580b225f6adba3e4f885eb4965761ac6501e23c8e52953b670478542bcdcb',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-U1-x86_64-Live-MATE.iso'},
+ {'name': 'RFRemix 20 U1 x86_64 XFCE',
+  'sha256': 'abc837386deaf46cb0d6d362ac6075935b844769aebd143a85e0df0b9ba4d439',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-U1-x86_64-Live-XFCE.iso'},
+ {'name': 'RFRemix 20 x86_64 GNOME',
+  'sha256': '6010608cd428d623d597fb2c463c3ae9ab912824bbc6eca7dc5a307b9c7d039a',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-x86_64-Live-GNOME.iso'},
+ {'name': 'RFRemix 20 x86_64 KDE',
+  'sha256': 'ef29e4f7a7be0f8f1b647978f2e96b7e3f549b1ca88f4050e977b4d17795a6fc',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-x86_64-Live-KDE.iso'},
+ {'name': 'RFRemix 20 x86_64 KDE minimal',
+  'sha256': '732ab32229101b8699bf942711b5d541113ac6bcd30ad8e488ceafefa9d7336f',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-x86_64-Live-KDE-minimal.iso'},
+ {'name': 'RFRemix 20 x86_64 LXDE',
+  'sha256': '90825d5c5b30d81b54645e14a6ff9734e342e0b7ea2f035eb80a0506fd5afa11',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-x86_64-Live-LXDE.iso'},
+ {'name': 'RFRemix 20 x86_64 MATE',
+  'sha256': 'e13a69f01347fedfa2e9067f906dae0dffa4908784437bd7faf3e7bfad19ccd5',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-x86_64-Live-MATE.iso'},
+ {'name': 'RFRemix 20 x86_64 XFCE',
+  'sha256': 'c1318c5bed8a6e0576f09ae9ed85bde6702716dcc984c498932ba07cf3f95745',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/x86_64/RFRemix-20-x86_64-Live-XFCE.iso'},
+ {'name': 'RFRemix 20 U1 i686 GNOME',
+  'sha256': 'deed40f3fcdd3572070a8c337a75f9fd8ca5c6adb6c33ff2c1f0b7dc1421435c',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-U1-i686-Live-GNOME.iso'},
+ {'name': 'RFRemix 20 U1 i686 KDE',
+  'sha256': 'f4a72b6e41e1a8d30e6675aa02ff84a5c47ba800c170357327856e5692f40b90',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-U1-i686-Live-KDE.iso'},
+ {'name': 'RFRemix 20 U1 i686 KDE minimal',
+  'sha256': 'e95c5d2d84c15458cb11c6a8bd515499b79c806eee105d40e38b1383444d61f4',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-U1-i686-Live-KDE-minimal.iso'},
+ {'name': 'RFRemix 20 U1 i686 LXDE',
+  'sha256': 'aabd5d8820b75ecbd3baa52fa3ecb1079183b2cba8ba97c23320cc6c2f8eaf9f',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-U1-i686-Live-LXDE.iso'},
+ {'name': 'RFRemix 20 U1 i686 MATE',
+  'sha256': '95d1d54e5883e327ebf5d0218665feaa78c093b0a14b655b50feb4bef4662703',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-U1-i686-Live-MATE.iso'},
+ {'name': 'RFRemix 20 U1 i686 XFCE',
+  'sha256': 'eac457b10e26893a4116921f28aac9ee377455792b90efdf0766c448758f0a8c',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-U1-i686-Live-XFCE.iso'},
+ {'name': 'RFRemix 20 i686 GNOME',
+  'sha256': 'f873a2c26012f42ce9bc8e05dec16f7802a78e720e22941e08c452cbaabb3732',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-i686-Live-GNOME.iso'},
+ {'name': 'RFRemix 20 i686 KDE',
+  'sha256': '79785e0ca454c327f094fda2644561a8085258a997d6aaeb00588bcbe32262cf',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-i686-Live-KDE.iso'},
+ {'name': 'RFRemix 20 i686 KDE minimal',
+  'sha256': '788a112fda00e565e261cba5567dae029710d81ace77a719a02b1ccdb2e1f619',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-i686-Live-KDE-minimal.iso'},
+ {'name': 'RFRemix 20 i686 LXDE',
+  'sha256': 'b0bef2842eb0b052d66f951456a2cf25db730d18d442988977acb14ae8f5ad56',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-i686-Live-LXDE.iso'},
+ {'name': 'RFRemix 20 i686 MATE',
+  'sha256': '512fa3493fd8d8d46933581589fb8d74fa532699e1a868a3278e17d2d9064de9',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-i686-Live-MATE.iso'},
+ {'name': 'RFRemix 20 i686 XFCE',
+  'sha256': '14edcda66895cd82ebc506aea7d6e5faba9410f3cb58946401345b3068dbee48',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/20/Live/i686/RFRemix-20-i686-Live-XFCE.iso'},
+ {'name': 'RFRemix 19.1 x86_64 GNOME',
+  'sha256': 'fd1cd7f2f2d665e35ff1e225915e6f2745d1ad69a6ffbd9d580a9b1041401daa',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/x86_64/RFRemix-19.1-x86_64-Live-GNOME.iso'},
+ {'name': 'RFRemix 19.1 x86_64 KDE',
+  'sha256': '28dee74ced84d4b9e4d1d46d11d23799e8a2db9b8d7b78c020a6d710360c3226',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/x86_64/RFRemix-19.1-x86_64-Live-KDE.iso'},
+ {'name': 'RFRemix 19.1 x86_64 KDE minimal',
+  'sha256': 'a5860b48d242fcf5a6fd376ae06fc8f142b4a543a746d10b1b604b3a093285d8',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/x86_64/RFRemix-19.1-x86_64-Live-KDE-minimal.iso'},
+ {'name': 'RFRemix 19.1 x86_64 LXDE',
+  'sha256': '481f690ba41cc0545867d5e542edd3d08c109517f65f24b62191597e4ba2b580',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/x86_64/RFRemix-19.1-x86_64-Live-LXDE.iso'},
+ {'name': 'RFRemix 19.1 x86_64 MATE',
+  'sha256': 'c3c1b5c1ef6d3eabbca72588c14433055b41d2d82e5f59d29efacd06841ee42f',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/x86_64/RFRemix-19.1-x86_64-Live-MATE.iso'},
+ {'name': 'RFRemix 19.1 x86_64 XFCE',
+  'sha256': 'bd6d1d1b99304754f9a607b27a8bf5c4c2e6a48801cfd64c8cddb4b8b0b762a6',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/x86_64/RFRemix-19.1-x86_64-Live-XFCE.iso'},
+ {'name': 'RFRemix 19.1 i686 GNOME',
+  'sha256': 'd96798cdb28cc67a028018ba335a9600be03354f0d5654407a0e508a1784ef53',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/i686/RFRemix-19.1-i686-Live-GNOME.iso'},
+ {'name': 'RFRemix 19.1 i686 KDE',
+  'sha256': 'e940848228b213d7ab73ffd1a5e974b801ece70cc37e924775b82b29d44a6794',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/i686/RFRemix-19.1-i686-Live-KDE.iso'},
+ {'name': 'RFRemix 19.1 i686 KDE minimal',
+  'sha256': 'e537a776cb57a4343248a3501f62446329a1a7df801c8ad2c4e062456dc93d04',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/i686/RFRemix-19.1-i686-Live-KDE-minimal.iso'},
+ {'name': 'RFRemix 19.1 i686 LXDE',
+  'sha256': 'ddb2b09bba510c42609d54a0cc4d3ea954a29badd0b7d1ee552c08a4834f926f',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/i686/RFRemix-19.1-i686-Live-LXDE.iso'},
+ {'name': 'RFRemix 19.1 i686 MATE',
+  'sha256': '8308d937f20cec801ce0f19fa6ec35a351eae71b774e3f823b638e0731c8a723',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/i686/RFRemix-19.1-i686-Live-MATE.iso'},
+ {'name': 'RFRemix 19.1 i686 XFCE',
+  'sha256': 'ddbcfec3ae7fd7651a6f781a69dd83ddbe34bb6564c7dd6dc98eca7d8e3e31ff',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/19.1/Live/i686/RFRemix-19.1-i686-Live-XFCE.iso'},
+ {'name': 'RFRemix 18 x86_64 GNOME',
+  'sha256': '60b4d58095e8c4770b6279bb320c972d71ed43700304f96ac3ca5d90960739cc',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/x86_64/RFRemix-18-x86_64-Live-GNOME.iso'},
+ {'name': 'RFRemix 18 x86_64 KDE',
+  'sha256': '8deb565fa994a809bb7edcf76e379d12f7833b0d3c25ddad7ce96608c04bb5bc',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/x86_64/RFRemix-18-x86_64-Live-KDE.iso'},
+ {'name': 'RFRemix 18 x86_64 KDE minimal',
+  'sha256': 'f69fef6c16678d5bc6859c0a33be7f589669b1fc68d7719714ca2dd1dec61709',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/x86_64/RFRemix-18-x86_64-Live-KDE-minimal.iso'},
+ {'name': 'RFRemix 18 x86_64 LXDE',
+  'sha256': 'ecb20550edf72b0fd68d93b36829b939ed798e825a3e50dc06a2dd94d2aa5fd5',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/x86_64/RFRemix-18-x86_64-Live-LXDE.iso'},
+ {'name': 'RFRemix 18 x86_64 MATE',
+  'sha256': '967bd7cb4dd408063b2aaf5a49e3756406e8206de764a015a077fe1dcd198887',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/x86_64/RFRemix-18-x86_64-Live-MATE.iso'},
+ {'name': 'RFRemix 18 x86_64 XFCE',
+  'sha256': 'af6a36de479cccb0743433e6b512e53b85843565e1fc442d3979fca09119cd19',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/x86_64/RFRemix-18-x86_64-Live-XFCE.iso'},
+ {'name': 'RFRemix 18 i686 GNOME',
+  'sha256': 'dbc6341ff05a95a4c4607c962fa588adf9c9f71c838ae01729045c5c51ce4675',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/i686/RFRemix-18-i686-Live-GNOME.iso'},
+ {'name': 'RFRemix 18 i686 KDE',
+  'sha256': '32a297e707676aff01083492a8a26022fc66e4cff7792c0a0983219b7f68ae20',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/i686/RFRemix-18-i686-Live-KDE.iso'},
+ {'name': 'RFRemix 18 i686 KDE minimal',
+  'sha256': '16a7aaa1d9fa849050376bf9a83ea2e2e9f6e52e1da91829afcc2c1f6939faf3',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/i686/RFRemix-18-i686-Live-KDE-minimal.iso'},
+ {'name': 'RFRemix 18 i686 LXDE',
+  'sha256': 'a3ac8f23570f1bfec2e132ec7b6274ae7cc2420cb41a855ae2125b294d417755',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/i686/RFRemix-18-i686-Live-LXDE.iso'},
+ {'name': 'RFRemix 18 i686 MATE',
+  'sha256': '71288cd67772001ea74c1bf9f2404d4c79e02a6ee67d374a8606caea44d47017',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/i686/RFRemix-18-i686-Live-MATE.iso'},
+ {'name': 'RFRemix 18 i686 XFCE',
+  'sha256': 'decba4be40c64e8eef7ab138251cf4f350b1e62ea39cc3a78ceae5361ce49998',
+  'url': 'http://mirror.yandex.ru/fedora/russianfedora/releases/RFRemix/18/Live/i686/RFRemix-18-i686-Live-XFCE.iso'},
+ {'name': 'Fedora 20 Desktop x86_64',
+  'sha256': 'cc0333be93c7ff2fb3148cb29360d2453f78913cc8aa6c6289ae6823372a77d2',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/x86_64/Fedora-Live-Desktop-x86_64-20-1.iso'},
+ {'name': 'Fedora 20 KDE x86_64',
+  'sha256': '08360a253b4a40dff948e568dba1d2ae9d931797f57aa08576b8b9f1ef7e4745',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/x86_64/Fedora-Live-KDE-x86_64-20-1.iso'},
+ {'name': 'Fedora 20 LXDE x86_64',
+  'sha256': 'b5002a697ef0e9e6fe10d0b88da6f7d43dbeb1b2c6dccb274b019123f321487d',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/x86_64/Fedora-Live-LXDE-x86_64-20-1.iso'},
+ {'name': 'Fedora 20 MATE Compiz x86_64',
+  'sha256': '37a3670955210b11e25af93548e1709973431b385379399952de6ae50567b8aa',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/x86_64/Fedora-Live-MATE-Compiz-x86_64-20-1.iso'},
+ {'name': 'Fedora 20 SoaS x86_64',
+  'sha256': 'b1865e40e57ed5c4bb705f95e3190c5e56ca6ed9c34f53b16e8c45ccb1233be8',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/x86_64/Fedora-Live-SoaS-x86_64-20-1.iso'},
+ {'name': 'Fedora 20 Xfce x86_64',
+  'sha256': 'ebfe836aa708d38b66a7ae6fe685ef327772ece5700861bc7c4e83baef0ceb1b',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/x86_64/Fedora-Live-Xfce-x86_64-20-1.iso'},
+ {'name': 'Fedora 20 Desktop i686',
+  'sha256': 'b115c5653b855de2353e41ff0c72158350f14a020c041462f35ba2a47bd1e33b',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/i386/Fedora-Live-Desktop-i686-20-1.iso'},
+ {'name': 'Fedora 20 KDE i686',
+  'sha256': 'd859132ea9496994ccbb5d6e60c9f40ae89ba31f8a4a1a2a883d6d45901de598',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/i386/Fedora-Live-KDE-i686-20-1.iso'},
+ {'name': 'Fedora 20 LXDE i686',
+  'sha256': 'cdfdaa74946792c3e500e32ac923f35c5a6730add9fa1c0997b7ac9b1f7cecae',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/i386/Fedora-Live-LXDE-i686-20-1.iso'},
+ {'name': 'Fedora 20 MATE Compiz i686',
+  'sha256': 'f885e1bf4db47c2093f0dfa9ef8861bd47103972e3be69ee47054d2568dacd67',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/i386/Fedora-Live-MATE-Compiz-i686-20-1.iso'},
+ {'name': 'Fedora 20 SoaS i686',
+  'sha256': 'efe76d842a7e19a5ee66461bf51a9cf649d426d007841a5f06f14851dceab389',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/i386/Fedora-Live-SoaS-i686-20-1.iso'},
+ {'name': 'Fedora 20 Xfce i686',
+  'sha256': '7d3c38414e85c956ea8ead7a1ca333559cec9252e6c330794215ad87a9829b77',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/20/Live/i386/Fedora-Live-Xfce-i686-20-1.iso'},
+ {'name': 'Fedora 19 Desktop x86_64',
+  'sha256': '21f0197284b9088b32d683f83a71bd42261f1df885a63b1eb87254d1ca096f12',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/19/Live/x86_64/Fedora-Live-Desktop-x86_64-19-1.iso'},
+ {'name': 'Fedora 19 KDE x86_64',
+  'sha256': 'f4479d5639f62d7398722c6a432ba94711b80d5011d5b64e7afebb4f4ac10cf7',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/19/Live/x86_64/Fedora-Live-KDE-x86_64-19-1.iso'},
+ {'name': 'Fedora 19 Desktop i686',
+  'sha256': 'ce9797802ef1f7aa670fffd04f209631c171b8ded5dc26f61df898cb9441c839',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/19/Live/i386/Fedora-Live-Desktop-i686-19-1.iso'},
+ {'name': 'Fedora 19 KDE i686',
+  'sha256': 'e81717564f96ee0f8c6d8b0186e7fca44da57f2ef154ea427b6d765ea139a083',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/19/Live/i386/Fedora-Live-KDE-i686-19-1.iso'},
+ {'name': 'Fedora 18 x86_64 Desktop',
+  'sha256': 'a276e06d244e04b765f0a35532d9036ad84f340b0bdcc32e0233a8fbc31d5bed',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/18/Live/x86_64/Fedora-18-x86_64-Live-Desktop.iso'},
+ {'name': 'Fedora 18 x86_64 KDE',
+  'sha256': '41d51d86ff5c272263285d00a0c3da7acbbce404b9930b0ff8bd7226e7248805',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/18/Live/x86_64/Fedora-18-x86_64-Live-KDE.iso'},
+ {'name': 'Fedora 18 i686 Desktop',
+  'sha256': '7c7f453c15a5d13df95bf8caab6277e5aab1c6353eb242b1cf00344b61869d26',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/18/Live/i386/Fedora-18-i686-Live-Desktop.iso'},
+ {'name': 'Fedora 18 i686 KDE',
+  'sha256': 'f172192566d0e12c29a741a568a917c5d8643aa781fdf06598b12a217a58cb74',
+  'url': 'http://dl.fedoraproject.org/pub/fedora/linux/releases/18/Live/i386/Fedora-18-i686-Live-KDE.iso'}
 ]
 
 other_releases = [
@@ -52,41 +233,57 @@ other_releases = [
 releases = fedora_releases + other_releases
 
 
-def get_fedora_releases():
-    global releases
-    fedora_releases = []
-    try:
-        html = urlread(FEDORA_RELEASES)
-        for release in re.findall(r'<a href="(\d+)/">', html)[-2:][::-1]:
-            for arch in ARCHES:
-                arch_url = FEDORA_RELEASES + '%s/Live/%s/' % (release, arch)
-                try:
-                    files = urlread(arch_url)
-                except URLGrabError:
-                    continue
-                for link in re.findall(r'<a href="(.*)">', files):
-                    if link.endswith('-CHECKSUM'):
-                        checksum = urlread(arch_url + link)
-                        for line in checksum.split('\n'):
-                            try:
-                                sha256, filename = line.split()
-                                if filename[0] != '*':
-                                    continue
+def get_fedora_releases(url):
+    releases = []
+    html = urlread(url)
+
+    # RFRemix distributions can have '.' in version.
+    # This code chooses the latest version for three last distributions.
+    # It's mean for 18, 17.1, 17, 16.1, 16, 15.1 and so on it will be 18, 17.1 and 16.1
+
+    # This code is dirty. Feel to free improve it.
+    all_releases = re.findall(r'<a href="(\d+\.?\d?)/">', html)[::-1]
+    target_releases = all_releases[:1]
+    last_release = int(target_releases[0].split('.')[0])
+    prev_release = last_release - 1
+    for release in all_releases:
+        if int(release.split('.')[0]) == prev_release:
+            target_releases.append(release)
+            prev_release = prev_release - 1
+        if prev_release == last_release - 3:
+            break
+
+    for release in target_releases:
+        for arch in ARCHES:
+            arch_url = url + '%s/Live/%s/' % (release, arch)
+            try:
+                files = urlread(arch_url)
+            except URLGrabError:
+                continue
+            for link in re.findall(r'<a href="(.*)">', files):
+                if link.endswith('-CHECKSUM'):
+                    checksum = urlread(arch_url + link)
+                    for line in checksum.split('\n'):
+                        try:
+                            sha256, filename = line.split()
+                            if filename.endswith('.iso') == False:
+                                continue
+
+                            if filename[0] == '*':
                                 filename = filename[1:]
-                                chunks = filename[:-6].split('-')
-                                chunks.remove('Live')
+
+                            chunks = filename[:-4].split('-')
+                            chunks.remove('Live')
+                            if chunks[-1] == '1':
+                                chunks.pop()
                                 release = chunks.pop()
                                 chunks.insert(1,release)
-                                name = ' '.join(chunks)
-                                fedora_releases.append(dict(
-                                    name=name,
-                                    url=arch_url + filename,
-                                    sha256=sha256,
+                            name = ' '.join(chunks)
+                            releases.append(dict(
+                                name = name,
+                                url = arch_url + filename,
+                                sha256 = sha256,
                                 ))
-                            except ValueError:
-                                pass
-        releases = fedora_releases + other_releases
-    except:
-        # Can't fetch releases from the internet.
-        releases += other_releases
+                        except ValueError:
+                            pass
     return releases
