diff -uNdr liveusb-creator-3.11.0.orig/liveusb/creator.py liveusb-creator-3.11.0/liveusb/creator.py
--- liveusb-creator-3.11.0.orig/liveusb/creator.py	2011-04-26 05:03:58.000000000 +0400
+++ liveusb-creator-3.11.0/liveusb/creator.py	2011-05-26 18:06:41.789350423 +0400
@@ -447,7 +447,7 @@
                 data = {
                     'udi': str(device),
                     'is_optical': bool(dev.Get(device, 'DeviceIsOpticalDisc')),
-                    'label': str(dev.Get(device, 'IdLabel')).replace(' ', '_'),
+                    'label': unicode(dev.Get(device, 'IdLabel')).replace(' ', '_'),
                     'fstype': str(dev.Get(device, 'IdType')),
                     'fsversion': str(dev.Get(device, 'IdVersion')),
                     'uuid': str(dev.Get(device, 'IdUuid')),
